include/nbl/builtin/hlsl/device_capabilities_traits.hlsl

```diff
@@ -25,22 +484,817 @@ namespace hlsl
+NBL_GENERATE_GET_OR_DEFAULT(performanceCounterMultipleQueryPools, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(mixedAttachmentSamples, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(accelerationStructure, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(accelerationStructureIndirectBuild, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(accelerationStructureHostCommands, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rayTracingPipeline, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rayTraversalPrimitiveCulling, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rayQuery, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(representativeFragmentTest, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentDensityMap, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentDensityMapDynamic, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentDensityMapNonSubsampledImages, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(deviceCoherentMemory, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentShaderSampleInterlock, bool, false);
 NBL_GENERATE_GET_OR_DEFAULT(fragmentShaderPixelInterlock, bool, false);
-NBL_GENERATE_GET_OR_DEFAULT(maxOptimallyResidentWorkgroupInvocations, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentShaderShadingRateInterlock, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rectangularLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(bresenhamLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(smoothLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(stippledRectangularLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(stippledBresenhamLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(stippledSmoothLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(indexTypeUint8, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(deferredHostOperations, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(pipelineExecutableInfo, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(deviceGeneratedCommands, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rayTracingMotionBlur, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rayTracingMotionBlurPipelineTraceRaysIndirect, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentDensityMapDeferred, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rasterizationOrderColorAttachmentAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rasterizationOrderDepthAttachmentAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(rasterizationOrderStencilAttachmentAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(cooperativeMatrixRobustBufferAccess, bool, false);
+
+
+
+
+//limits
+NBL_GENERATE_GET_OR_DEFAULT(maxImageDimension1D, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxImageDimension2D, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxImageDimension3D, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxImageDimensionCube, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxImageArrayLayers, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxBufferViewTexels, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxUBOSize, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(MaxSSBOSize, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxPushConstantsSize, uint32_t, 0);
```

Comment:
pay attention to match the types in SPhysicalDeviceLimits.h

Reply:
ouch, also the default values... they can't all be 0

We'll eventually write a JSON + python based generator for these three headers